File: BCRPData.R
# BCRPData ----

## Librerias ----

#install.packages("httr")
library(httr)
#install.packages("jsonlite")
library(jsonlite)

## Metadatos de series ---

metadata <- read.csv(url("https://estadisticas.bcrp.gob.pe/estadisticas/series/metadata"),
                     sep=";")

## Funcion para consultar datos ----

bcrp_data <- function(serie, inicio, fin){
  
  # Establecer consulta
  periodo <- paste("/", inicio, "/", fin, sep="")
  url <- paste("https://estadisticas.bcrp.gob.pe/estadisticas/series/api/",
               serie,
               "/json",
               periodo,
               sep="")
  
  # Consulta de datos
  serie_api <- GET(url)
  serie_json <- fromJSON(rawToChar(serie_api$content))
  serie_df <- as.data.frame(serie_json$periods)
  names(serie_df)[1] <- "Fecha"
  names(serie_df)[2] <- as.character(serie_json$config$title)
  return(serie_df)
  
}

## Ejemplo ----

fecha_inicio <- "2017-01"
fecha_fin <- "2022-08"
pbi <- "PN01770AM"

bcrp_data(pbi, fecha_inicio, fecha_fin)

File: BCRPData.py
# -*- coding: utf-8 -*-
# BCRPData

## Librerias

import pandas as pd
import requests

## Metadatos de series

metadata = pd.read_csv('https://estadisticas.bcrp.gob.pe/estadisticas/series/metadata',
                       delimiter=';',
                       encoding='latin-1')

## Funcion para consultar datos

def bcrp_data(serie, inicio, fin):
  
  # Establecer consulta
  periodo = '/'+inicio+'/'+fin
  url = 'https://estadisticas.bcrp.gob.pe/estadisticas/series/api/'+serie+'/json'+periodo
  
  # Consulta de datos
  serie_api = requests.get(url)
  serie_json = serie_api.json()
  
  serie = serie_json.get('config')['series'][0]['name']
  periodos = serie_json.get('periods')
  
  valores = []
  for p in periodos:
    values = p['values']
    for v in values:
      v = float(v)
      valores.append(v)
  fechas = []
  for p in periodos:
    name = p['name']
    fechas.append(name)
  
  serie_dic = {"Fecha":fechas, str(serie):valores}
  serie_df = pd.DataFrame(serie_dic)
  return(serie_df)

## Ejemplo

fecha_inicio = '2017-01'
fecha_fin = '2022-08'
pbi = 'PN01770AM'

bcrp_data(pbi, fecha_inicio, fecha_fin)
